hnn_core/cell.py

```diff
@@ -228,28 +233,38 @@ def insert_dipole(self, yscale):
             dpp.ztan = y_diff[-1]
         self.dipole = h.Vector().record(self.dpl_ref)
 
-    def record_current_soma(self):
-        """Record current at soma."""
+    def record_soma(self, record_vsoma=False, record_isoma=False):
+        """Record current and voltage at soma.
+
+        Parameters
+        ----------
+        record_vsoma : bool
+            Option to record somatic voltages from cells
+            net.params['N_trials'] will be used
+        record_isoma : bool
+            Option to record somatic currents from cells
+
+        """
         # a soma exists at self.soma
-        try:
-            # assumes that self.synapses is a dict that exists
-            list_syn_soma = [key for key in self.synapses.keys()
-                             if key.startswith('soma_')]
-            # matching dict from the list_syn_soma keys
-            self.dict_currents = dict.fromkeys(list_syn_soma)
-            # iterate through keys and record currents appropriately
-            for key in self.dict_currents:
-                self.dict_currents[key] = h.Vector()
-                self.dict_currents[key].record(self.synapses[key]._ref_i)
-        except:
-            print(
-                "Warning in Cell(): record_current_soma() was called,"
-                " but no self.synapses dict was found")
-            pass
-
-    def record_voltage_soma(self):
-        """Record voltage at soma."""
-        self.rec_v.record(self.soma(0.5)._ref_v)
+        if record_isoma:
+            try:
```

Comment:
would something bad happen if you remove the `try`-`except` ? If not, I'd get rid of it

Reply:
I don't think so... this is code left over from the original current recording setup. If nothing changes I'll convert it to mimic the voltage recording naming schemes (including the variable name changes below).